classes/submission/reviewAssignment/ReviewAssignment.inc.php

```diff
@@ -464,6 +453,129 @@ function setReviewFormId($reviewFormId) {
+				return REVIEW_ASSIGNMENT_STATUS_RECEIVED;
+			}
+			return REVIEW_ASSIGNMENT_STATUS_THANKED;
+		} elseif ($this->getDateCompleted()) { // review submitted...
+			if ($this->getUnconsidered() != REVIEW_ASSIGNMENT_UNCONSIDERED && $this->isRead()) { // ...and confirmed by an editor
+				return REVIEW_ASSIGNMENT_STATUS_COMPLETE;
+			}
+			return REVIEW_ASSIGNMENT_STATUS_RECEIVED;
+		}
+
+		return REVIEW_ASSIGNMENT_STATUS_AWAITING_RESPONSE;
+	}
+
+	/**
+	 * Determine whether an editorial user has read this review
+	 *
+	 * @return bool
+	 */
+	function isRead() {
+		$submissionDao = Application::getSubmissionDAO();
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO');
+		$userStageAssignmentDao = DAORegistry::getDAO('UserStageAssignmentDAO');
+		$viewsDao = DAORegistry::getDAO('ViewsDAO');
+
+		$submission = $submissionDao->getById($this->getSubmissionId());
+
+		// Get the user groups for this stage
+		$userGroups = $userGroupDao->getUserGroupsByStage(
+			$submission->getContextId(),
+			$this->getStageId()
+		);
+		while ($userGroup = $userGroups->next()) {
+			$roleId = $userGroup->getRoleId();
+			if ($roleId != ROLE_ID_MANAGER && $roleId != ROLE_ID_SUB_EDITOR) {
+				continue;
+			}
+
+			// Get the users assigned to this stage and user group
+			$stageUsers = $userStageAssignmentDao->getUsersBySubmissionAndStageId(
+				$this->getSubmissionId(),
+				$this->getStageId(),
+				$userGroup->getId()
+			);
+
+			// Check if any of these users have viewed it
+			while ($user = $stageUsers->next()) {
+				if ($viewsDao->getLastViewDate(
+					ASSOC_TYPE_REVIEW_RESPONSE,
+					$this->getId(),
+					$user->getId()
+				)) {
+					return true;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get the translation key for the current status
+	 *
+	 * @param int $status Optionally pass a status to retrieve a specific key.
+	 *  Default will return the key for the current status.
+	 * @return string
+	 */
+	public function getStatusKey($status = null) {
+
+		if (is_null($status)) {
+			$status = $this->getStatus();
+		}
+
+		switch ($status) {
+			case REVIEW_ASSIGNMENT_STATUS_AWAITING_RESPONSE:
+				return 'submission.review.status.awaitingResponse';
+			case REVIEW_ASSIGNMENT_STATUS_DECLINED:
+				return 'submission.review.status.declined';
+			case REVIEW_ASSIGNMENT_STATUS_RESPONSE_OVERDUE:
+				return 'submission.review.status.responseOverdue';
+			case REVIEW_ASSIGNMENT_STATUS_REVIEW_OVERDUE:
+				return 'submission.review.status.reviewOverdue';
+			case REVIEW_ASSIGNMENT_STATUS_ACCEPTED:
+				return 'submission.review.status.accepted';
+			case REVIEW_ASSIGNMENT_STATUS_RECEIVED:
+				return 'submission.review.status.received';
+			case REVIEW_ASSIGNMENT_STATUS_COMPLETE:
+				return 'submission.review.status.complete';
+			case REVIEW_ASSIGNMENT_STATUS_THANKED:
+				return 'submission.review.status.thanked';
+		}
+
+		error_log('No status key could be found for ' . get_class($this) . ' on ' . __LINE__);
```

Comment:
`error_log` messages will probably get lost/ignored; I'd favour either `assert` (for us devs to watch for) or `fatalError` (our long-established pattern of error handling) or an `Exception` (which is worth introducing as a normal habit now).